MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2211,85 +2211,81 @@ public string GetAudioFilterParam(EncodingJobInfo state, EncodingOptions encodin
 
             var request = state.BaseRequest;
 
-            var inputChannels = audioStream.Channels;
+            var codec = outputAudioCodec ?? string.Empty;
 
-            if (inputChannels <= 0)
-            {
-                inputChannels = null;
-            }
+            int? resultChannels = state.GetRequestedAudioChannels(codec);
 
-            var codec = outputAudioCodec ?? string.Empty;
+            var inputChannels = audioStream.Channels;
 
-            int? transcoderChannelLimit;
-            if (codec.IndexOf("wma", StringComparison.OrdinalIgnoreCase) != -1)
-            {
-                // wmav2 currently only supports two channel output
-                transcoderChannelLimit = 2;
-            }
-            else if (codec.IndexOf("mp3", StringComparison.OrdinalIgnoreCase) != -1)
-            {
-                // libmp3lame currently only supports two channel output
-                transcoderChannelLimit = 2;
-            }
-            else if (codec.IndexOf("aac", StringComparison.OrdinalIgnoreCase) != -1)
-            {
-                // aac is able to handle 8ch(7.1 layout)
-                transcoderChannelLimit = 8;
-            }
-            else
+            if (inputChannels > 0)
             {
-                // If we don't have any media info then limit it to 6 to prevent encoding errors due to asking for too many channels
-                transcoderChannelLimit = 6;
+                resultChannels = inputChannels < resultChannels ? inputChannels : resultChannels ?? inputChannels;
             }
 
             var isTranscodingAudio = !IsCopyCodec(codec);
 
-            int? resultChannels = state.GetRequestedAudioChannels(codec);
             if (isTranscodingAudio)
             {
-                resultChannels = GetMinValue(request.TranscodingMaxAudioChannels, resultChannels);
-            }
-
-            if (inputChannels.HasValue)
-            {
-                resultChannels = resultChannels.HasValue
-                    ? Math.Min(resultChannels.Value, inputChannels.Value)
-                    : inputChannels.Value;
-            }
+                // Set max transcoding channels to 8 to prevent encoding errors due to asking for too many channels
+                // AAC, FLAC, ALAC, libopus, libvorbis encoders all support at least 8 channels
+                int transcoderChannelLimit = 8;
```

Comment:
Please cap the max channel to 6 on `libfdk_aac`. Or it produces broken audio in HLS streaming.

Reply:
I've switched to querying for the encoder vs. the codec so we can be more precise. Added libfdk_aac and a few others.